### PR TITLE
Add aten mkldnn add operator

### PR DESCRIPTION
**Pull request by bddppq**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19633 Add is_mkldnn to at::Tensor&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D15053320/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19204 Add aten mkldnn conv2d operator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14857513/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19205 Add aten mkldnn ops: relu, max_pool2d and avg_pool2d&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14850598/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19206 Add aten mkldnn batch_norm operator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14887205/)
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#19207 Add aten mkldnn add operator**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14889477/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19209 Add aten mkldnn view operator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14894545/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19210 Add aten mkldnn linear operator&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14901641/)
&nbsp;&nbsp;&nbsp;&nbsp;:white_circle:&nbsp; #19648 Adjust resnext run script&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D14889481/)

Pull Request resolved: https://github.com/pytorch/pytorch/pull/19207

Differential Revision: [D14889477](https://our.internmc.facebook.com/intern/diff/D14889477/)

### CODE REVIEW

File: aten/src/ATen/native/native_functions.yaml [hunk 1]
```diff
@@ -128,11 +128,30 @@
...
   variants: method
+  dispatch:
+    CPU: add_
+    CUDA: add_
+    SparseCPU: add_
+    SparseCUDA: add_
+    MkldnnCPU: mkldnn_add_
 
 - func: add(Tensor self, Tensor other, *, Scalar alpha=1, Tensor(a!) out) -> Tensor(a!)
+  matches_jit_signature: True
```
Comment (dzhulgakov) [inline aten/src/ATen/native/native_functions.yaml]: no need to put matches_jit_signature

File: test/test_mkldnn.py [hunk 2]
```diff
@@ -160,6 +160,28 @@ def test_batch_norm2d(self):
...
+        x = torch.randn(N, C, 35, 45, dtype=torch.float32) * 10
+        y = torch.randn(N, C, 35, 45, dtype=torch.float32) * 10
+        mx = x.to_mkldnn()
+        my = y.to_mkldnn()
+
+        self.assertEqual(
+            x + y,
+            (mx + my).to_dense())
+
+        self.assertEqual(
```
Comment (dzhulgakov) [inline test/test_mkldnn.py]: I'd also test add_ and add_out variants to make sure we didn't screw up them

File: aten/src/ATen/native/mkldnn/BinaryOps.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+Tensor& mkldnn_add_out(
+    Tensor& result,
+    const Tensor& self,
+    const Tensor& other,
+    Scalar alpha) {
+  ideep::tensor& x = itensor_from_mkldnn(self);
+  ideep::tensor& y = itensor_from_mkldnn(other);
+
+  ideep::tensor& z = itensor_from_mkldnn(result);
+  const std::vector<float> scales{1.0, alpha.to<float>()};
```
Comment (gchanan) [inline aten/src/ATen/native/mkldnn/BinaryOps.cpp]: do we ever worry about mkldnn supporting double?  We could technically write this generically with a dispatch macro.

File: aten/src/ATen/native/native_functions.yaml (hunk 1 above)
Comment (gchanan) [reply aten/src/ATen/native/native_functions.yaml]: yes, please remove this, I just spent a bunch of time killing all of these :).

**Review by gchanan — APPROVED** [review]
approved once comments are addressed.

File: aten/src/ATen/native/mkldnn/BinaryOps.cpp (hunk 3 above)
Comment (bddppq) [reply aten/src/ATen/native/mkldnn/BinaryOps.cpp]: [ideep::sum::compute](https://bddppq.github.io/codebrowser/pytorch/pytorch/third_party/ideep/include/ideep/computations.hpp.html#_ZN5ideep3sum7computeERKSt6vectorIfSaIfEERKS1_INS_6tensorESaIS6_EERS6_) only takes [scales](https://bddppq.github.io/codebrowser/pytorch/pytorch/third_party/ideep/include/ideep/abstract_types.hpp.html#ideep::scale_t) as float vector

**Review by dzhulgakov — APPROVED** [review]

**Pull request closed without merging**